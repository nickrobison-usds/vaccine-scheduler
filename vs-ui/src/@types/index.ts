import Client from "fhir-kit-client";

export interface OperationParams<T extends fhir.DomainResource> {
    name: string;
    resourceType?: string;
    id?: string;
    method?: "get" | "post";
    input?: string | T;
}

export interface FhirClientResponse {
    status: number;
    data: fhir.OperationOutcome;
}

export interface FhirClientError {
    config: any; // Type this
    response: FhirClientResponse;
}

export interface OperationClient extends Client {
    operation<T extends fhir.DomainResource>(params: OperationParams<T>): Promise<fhir.OperationOutcome>;
}

// Guards

export function isClientError(err: any): err is FhirClientError {
    return (err as FhirClientError).response !== undefined;
}

export function isFHIRResource(json: any): json is fhir.DomainResource {
    return (json as fhir.DomainResource).resourceType !== undefined;
}

export function isBundle(bundle: any): bundle is fhir.Bundle {
    return (bundle as fhir.Bundle).type !== undefined;
}
