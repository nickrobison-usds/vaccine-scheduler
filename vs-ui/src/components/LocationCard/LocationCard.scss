.location-card {
  max-width: 90%;
}

.location-card--footer {
  padding-bottom: 0;
  padding-left: 0;
  padding-right: 0;
  text-align: right;
}
