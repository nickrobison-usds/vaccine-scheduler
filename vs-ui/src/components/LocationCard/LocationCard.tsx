import { faExclamationTriangle } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
  Card,
  CardBody,
  CardFooter,
  CardHeader,
} from "@trussworks/react-uswds";
import React from "react";
import "./LocationCard.scss";

export interface LocationCardProps {
  location: fhir.Location;
}

const LocationAddress: React.FC<{ address: fhir.Address }> = ({ address }) => {
  return (
    <div>
      {address.line![0]}
      <br />
      {address.city!}, {address.state!} {address.postalCode!}
    </div>
  );
};

export const LocationCard: React.FC<LocationCardProps> = ({ location }) => {
  const address = location.address;

  return (
    <Card className="location-card">
      <CardHeader>
        <h3 className="usa-card__heading">
          {location.name ? location.name : ""}
        </h3>
        <CardBody>{address && <LocationAddress address={address} />}</CardBody>
        <CardFooter className="location-card--footer">
          <FontAwesomeIcon icon={faExclamationTriangle} />
          <span />
          100 slots available
        </CardFooter>
      </CardHeader>
    </Card>
  );
};
