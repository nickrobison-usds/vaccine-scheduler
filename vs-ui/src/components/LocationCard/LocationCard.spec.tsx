import { render } from "@testing-library/react";
import React from "react";

import { CapitatedLocation } from "../../@types";

import { LocationCard } from "./LocationCard";

describe("LocationCare", () => {
  it("renders with available slots", () => {
    const location: CapitatedLocation = {
      id: "test-id",
      address: {
        line: ["1234 Test Avenue"],
        city: "Washington",
        state: "DC",
        postalCode: "20008",
      },
      capacity: 100,
    };
    const { container } = render(<LocationCard location={location} />);
    expect(container).toMatchSnapshot();
  });

  it("renders with no available slots", () => {
    const location: CapitatedLocation = {
      id: "test-id",
      address: {
        line: ["1234 Test Avenue"],
        city: "Washington",
        state: "DC",
        postalCode: "20008",
      },
      capacity: 0,
    };
    const { container } = render(<LocationCard location={location} />);
    expect(container).toMatchSnapshot();
  });

  it("renders with unknown slots", () => {
    const location: CapitatedLocation = {
      id: "test-id",
      address: {
        line: ["1234 Test Avenue"],
        city: "Washington",
        state: "DC",
        postalCode: "20008",
      },
    };
    const { container } = render(<LocationCard location={location} />);
    expect(container).toMatchSnapshot();
  });
});
