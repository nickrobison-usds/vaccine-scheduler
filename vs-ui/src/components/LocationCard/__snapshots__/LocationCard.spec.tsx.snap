// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationCare renders with available slots 1`] = `
<div>
  <li
    class="usa-card location-card"
    data-testid="Card"
  >
    <div
      class="usa-card__container"
    >
      <header
        class="usa-card__header"
        data-testid="CardHeader"
      >
        <h3
          class="usa-card__heading"
        />
      </header>
      <div
        class="usa-card__body"
        data-testid="CardBody"
      >
        <div>
          1234 Test Avenue
          <br />
          Washington
          , 
          DC
           
          20008
        </div>
      </div>
      <div
        class="usa-card__footer location-card--footer"
        data-testid="CardFooter"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-check-circle fa-w-16 "
          data-icon="check-circle"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
            fill="currentColor"
          />
        </svg>
        100 slots available
      </div>
    </div>
  </li>
</div>
`;

exports[`LocationCare renders with no available slots 1`] = `
<div>
  <li
    class="usa-card location-card"
    data-testid="Card"
  >
    <div
      class="usa-card__container"
    >
      <header
        class="usa-card__header"
        data-testid="CardHeader"
      >
        <h3
          class="usa-card__heading"
        />
      </header>
      <div
        class="usa-card__body"
        data-testid="CardBody"
      >
        <div>
          1234 Test Avenue
          <br />
          Washington
          , 
          DC
           
          20008
        </div>
      </div>
      <div
        class="usa-card__footer location-card--footer"
        data-testid="CardFooter"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-ban fa-w-16 "
          data-icon="ban"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 8C119.034 8 8 119.033 8 256s111.034 248 248 248 248-111.034 248-248S392.967 8 256 8zm130.108 117.892c65.448 65.448 70 165.481 20.677 235.637L150.47 105.216c70.204-49.356 170.226-44.735 235.638 20.676zM125.892 386.108c-65.448-65.448-70-165.481-20.677-235.637L361.53 406.784c-70.203 49.356-170.226 44.736-235.638-20.676z"
            fill="currentColor"
          />
        </svg>
        No slots available
      </div>
    </div>
  </li>
</div>
`;

exports[`LocationCare renders with unknown slots 1`] = `
<div>
  <li
    class="usa-card location-card"
    data-testid="Card"
  >
    <div
      class="usa-card__container"
    >
      <header
        class="usa-card__header"
        data-testid="CardHeader"
      >
        <h3
          class="usa-card__heading"
        />
      </header>
      <div
        class="usa-card__body"
        data-testid="CardBody"
      >
        <div>
          1234 Test Avenue
          <br />
          Washington
          , 
          DC
           
          20008
        </div>
      </div>
      <div
        class="usa-card__footer location-card--footer"
        data-testid="CardFooter"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-exclamation-circle fa-w-16 "
          data-icon="exclamation-circle"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
            fill="currentColor"
          />
        </svg>
        Unknown availability
      </div>
    </div>
  </li>
</div>
`;
